### PR TITLE
Initial implementation

### PR DESCRIPTION
**Pull request by hyperbolic2346**
A subordinate charm that allows using AWS IAM credentials to authenticate and authorize on your Charmed Kubernetes install.

Depends on:
https://github.com/charmed-kubernetes/charm-kubernetes-master/pull/44
https://github.com/charmed-kubernetes/layer-kubernetes-common/pull/3
https://github.com/juju-solutions/interface-aws-iam/pull/1

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+```
+juju deploy cs:~containers/aws-iam
+juju deploy charmed-kubernetes
+juju add-relation aws-iam kubernetes-master
+```
+
+## Further information
+
+- [AWS IAM Homepage](https://github.com/kubernetes-sigs/aws-iam-authenticator)
+- [AWS IAM Charm Issue Tracker]()
```
Comment (johnsca) [inline README.md]: This link is missing.

**Comment by johnsca** [comment]
LGTM save for the missing bug tracker link

**Comment by hyperbolic2346** [comment]
Added image as a configuration option and removed the hard-coded docker image. Please review those changes.

**Comment by hyperbolic2346** [comment]
https://bugs.launchpad.net/charm-kubernetes-master/+bug/1841439

**Pull request merged**